bindings/wasm/src/credential/revocation/status_list_2021/credential.rs

```diff
@@ -0,0 +1,221 @@
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(from = "StatusList2021Credential", into = "StatusList2021Credential")]
+pub struct WasmStatusList2021Credential {
+  pub(crate) inner: StatusList2021Credential,
+  wasm_credential: WasmCredential,
+}
+
+impl Deref for WasmStatusList2021Credential {
+  type Target = WasmCredential;
+  fn deref(&self) -> &Self::Target {
+    &self.wasm_credential
+  }
+}
+
+impl From<StatusList2021Credential> for WasmStatusList2021Credential {
+  fn from(value: StatusList2021Credential) -> Self {
+    Self {
+      wasm_credential: WasmCredential(value.clone().into_inner()),
+      inner: value,
+    }
+  }
+}
+
+impl From<WasmStatusList2021Credential> for StatusList2021Credential {
+  fn from(value: WasmStatusList2021Credential) -> Self {
+    value.inner
+  }
+}
+
+#[wasm_bindgen(js_class = StatusList2021Credential)]
+impl WasmStatusList2021Credential {
+  /// Creates a new {@link StatusList2021Credential}.
+  #[wasm_bindgen(constructor)]
+  pub fn new(credential: WasmCredential) -> Result<WasmStatusList2021Credential> {
+    StatusList2021Credential::try_from(credential.0)
+      .map(Into::into)
+      .map_err(|e| JsValue::from(JsError::new(&e.to_string())))
+  }
+
+  #[wasm_bindgen]
+  pub fn id(&self) -> String {
+    self.inner.id.as_deref().map(ToString::to_string).unwrap()
+  }
+
+  /// Sets the given credential's status using the `index`-th entry of this status list.
+  /// Returns the created `credentialStatus`.
+  #[wasm_bindgen(js_name = "setCredentialStatus")]
+  pub fn set_credential_status(
+    &mut self,
+    credential: &mut WasmCredential,
+    index: usize,
+    flag: bool,
+  ) -> Result<WasmStatusList2021Entry> {
+    let entry = self
+      .inner
+      .set_credential_status(&mut credential.0, index, flag)
+      .map_err(|e| JsValue::from(JsError::new(&e.to_string())))?
+      .clone();
+    self.wasm_credential = WasmCredential(self.inner.clone().into_inner());
+
+    Ok(WasmStatusList2021Entry(entry))
+  }
+
+  /// Returns the {@link StatusPurpose} of this {@link StatusList2021Credential}.
+  #[wasm_bindgen]
+  pub fn purpose(&self) -> WasmStatusPurpose {
+    self.inner.purpose().into()
+  }
+
+  /// Returns the state of the `index`-th entry, if any.
+  #[wasm_bindgen]
+  pub fn entry(&self, index: usize) -> Result<bool> {
+    self
+      .inner
+      .entry(index)
+      .map_err(|e| JsValue::from(JsError::new(&e.to_string())))
+  }
+
+  #[wasm_bindgen(js_name = "clone")]
+  pub fn wasm_clone(&self) -> WasmStatusList2021Credential {
+    self.clone()
+  }
+
+  #[wasm_bindgen(js_name = "fromJSON")]
+  pub fn from_json(json: JsValue) -> Result<WasmStatusList2021Credential> {
+    use crate::error::WasmResult;
+    json.into_serde::<WasmStatusList2021Credential>().wasm_result()
+  }
+
+  #[wasm_bindgen(js_name = "toJSON")]
+  pub fn to_json(&self) -> Result<JsValue> {
+    use crate::error::WasmResult;
+    JsValue::from_serde(self).wasm_result()
+  }
+}
```

Comment:
```suggestion
```
Use `impl_wasm_json` and `impl_wasm_clome` instead.

Reply:
Those macros only work with new type structs, cannot use them in this case.